x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.deprecation;
+
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.xpack.core.deprecation.DeprecationIssue;
+
+import java.util.Locale;
+
+public class NodeDeprecationChecks {
+
+    static DeprecationIssue checkRemovedSetting(final Settings settings, final Setting<?> removedSetting, final String url) {
+        if (removedSetting.exists(settings) == false) {
+            return null;
+        }
+        final String removedSettingKey = removedSetting.getKey();
+        final String value = removedSetting.get(settings).toString();
+        final String message =
+            String.format(Locale.ROOT, "setting [%s] is deprecated and will be removed in the next major version", removedSettingKey);
+        final String details =
+            String.format(Locale.ROOT, "the setting [%s] is currently set to [%s], remove this setting", removedSettingKey, value);
+        return new DeprecationIssue(DeprecationIssue.Level.CRITICAL, message, url, details);
```

Comment:
should this be `WARNING` instead of `CRITICAL` ?

If the setting is removed, won't it be be "archived" allowing the server to continue to function ? I think we are trying to reserve `CRITICAL` for things items that will prevent the server from running.

Reply:
Not necessarily, because these could be node level settings, which we don't archive, which do prevent startup if they are unrecognized. Also, we have discussing removing the archiving of settings, so this behavior could change for cluster-level settings too.

Reply:
In that case, should we flex WARNING or CRITICAL based if the setting has node scope ?

Also, should we check that that value is not the default value here so we don't issue an issue a deprecation issue for settings that user never changed ?

Reply:
There is a check that the setting is not set on line 19?

Reply:
ah .. i missed "Note that fallback settings are excluded." in the exists method.

Reply:
Yeah, that's why we have `Setting#existsOrFallbackExists`.